### DISCUSSION TITLE
Warn if the registration flow requires a registration token

### DISCUSSION THREAD

**Discussion by Half-Shot** [element-hq/element-meta#783] [Ideas]
### Your use case

Homeserver admins can configure their homeserver to require a registration token upon registration, a shared secret token that only allows some users to register. However, since Element isn't "aware" of this feature yet it doesn't prompt for the token on registration. Rather, it accepts the users registration details but then redirects them to the homeserver to enter the token.

This is frustrating for users who attempt to sign up to a host, only to be surprised by an additional requirement. Element is aware of all the registration flows, and could warn about future flows. My suggestion would be to implement a simple warning box when Element is going to redirect you to complete additional flows.

### Have you considered any alternatives?

Implement vector-im/element-web#7275, which should hopefully implicitly warn. Given the current progress however, it seems like a intermediary PR to warn might be appropriate.

### Additional context

As an example, a homeserver can be configured to have a set of registration flows like:

```json
{
    "session": "qeLMBdbowXJQqtCRvdlfSEID",
    "flows": [
        {
            "stages": [
                "m.login.registration_token",
                "m.login.terms",
                "m.login.email.identity"
            ]
        },
        {
            "stages": [
                "m.login.terms",
                "m.login.registration_token"
            ]
        }
    ],
    "params": {
        "m.login.terms": {
            "policies": {
                "privacy_policy": {
                    "version": "1.0",
                    "en": {
                        "name": "Privacy Policy",
                        "url": "https://foobar/_matrix/consent?v=1.0"
                    }
                }
            }
        }
    }
}
```

**Comment by MexHigh** [comment]
Why not implement a view for registration tokens like Element Web did lately? Element Android and iOS both fall back to the ugly webpages at `/_matrix/static/client/register` when the homeserver requires registration tokens.
